src/TypeScript/TsConfig/TsConfigProvider.cs

```diff
@@ -77,25 +77,33 @@ public async Task<string> GetConfigForFile(string sourceFilePath, CancellationTo
         {
             foreach (var tsConfigFilePath in candidateTsConfigs)
             {
-                if (visited.Contains(tsConfigFilePath))
+                var tsConfigToCheck = tsConfigFilePath;
+                var isDirectory = !Path.HasExtension(tsConfigToCheck);
+
+                if (isDirectory)
+                {
+                    tsConfigToCheck = Path.GetFullPath(Path.Combine(tsConfigToCheck, "tsconfig.json"));
+                }
+
+                if (visited.Contains(tsConfigToCheck))
```

Comment:
`Contains` does a case-sensitive comparison.
It would be better to use a hashset created with an appropriate comparer.